internal/coremsgs/en_error_messages.go

```diff
@@ -279,3 +278,8 @@ var (
 	MsgDeprecatedResetWithAutoReload      = ffe("FF10438", "The deprecated reset API cannot be used when dynamic config reload is enabled", 409)
 	MsgConfigArrayVsRawConfigMismatch     = ffe("FF10439", "Error processing configuration - mismatch between raw and processed array lengths")
 )
+
+// function versions of error factories that allow status code to be parameterized
+var MsgAddressResolveBadStatus = func(status int) i18n.ErrorMessageKey {
+	return ffe("FF10340", "Failed to resolve signing key string '%s' [%d]: %s", status)
+}
```

Comment:
Clever, but unfortunately I think this will cause problems. The `ffe` function doesn't build a specific instance of an error. It registers a specific error/translation code within the runtime. There is code that specifically checks to make sure the same code isn't already registered (helpful when making code changes and accidentally re-using an existing code).

I believe the code as written above will cause this function to panic:
https://github.com/hyperledger/firefly-common/blob/main/pkg/i18n/messages.go#L93-L95

Off the top of my head, I'm not sure of viable alternatives other than listing out the specific error codes we want to return, and then mapping to those. Or changing the way the `FFE/FFM` function works itself.

Reply:
good spot.  I just tested with a resolver URL of 
```
      addressResolver: 
        urlTemplate: "https://httpbin.org/status/{{.Key}}"
```
so that I could force different status codes and, as you predicted...
```
2023/01/31 18:50:25 http: panic serving 172.27.0.1:60446: Message ID FF10340 re-used
goroutine 754 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1825 +0xb8
panic({0xd3d740, 0x400045d0e0})
	/usr/local/go/src/runtime/panic.go:844 +0x260
github.com/hyperledger/firefly-common/pkg/i18n.FFM({0x2e08?, 0xac?, {0x0?, 0x0?}}, {0xe8781d, 0x7}, {0xebd096, 0x32})
	/go/pkg/mod/github.com/hyperledger/firefly-common@v1.2.1/pkg/i18n/messages.go:94 +0x284

```